sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -603,7 +603,7 @@ def pipeline_from_stages(
   components.transforms.clear()
   components.pcollections.clear()
 
-  roots = set()
+  roots = {}  # order preserving but still has fast contains checking
```

Comment:
Might it be worth adding a test for this?

Reply:
I thought a bit about this, not easy to set up a test for this specifically. This code is well exercised elsewhere though.